### ISSUE TITLE
Improve CodeCombat Travis CI build times

### ISSUE THREAD

**Issue by popey456963** [codecombat/codecombat#4335]
At the moment it's a staggering 8 minutes and 37 seconds with only 64 seconds for `jasmine` tests and 96 seconds for coverage tests!  Perhaps we should consider looking into ways of decreasing the time.

**Comment by duk3luk3** [comment]
Here's a spreadsheet with an artisanal hand-crafted natively-embedded flamegraph for https://travis-ci.org/codecombat/codecombat/builds/237406455:

![](http://i.imgur.com/OOMI7Wm.png)

Most of the runtime is taking up by npm.

Travis supports caching that, maybe? https://docs.travis-ci.com/user/caching/#Arbitrary-directories

**Comment by popey456963** [comment]
@duk3luk3 Such a beautiful flamegraph!  What I don't understand is why npm install is taking ~150 seconds, we do actually already make use of caching of the npm directory:

```yaml
cache:
  directories:
    - node_modules
    - bower_components
```

Not sure if npm@5 is the latest version yet (or it's still in beta), but that apparently has large speedups over npm@4.  Alternatively, perhaps shifting to Yarn might be a good idea?

**Comment by duk3luk3** [comment]
Yep, and I just checked the travis settings (since I got the permissions to do so now), and branching builds are enabled, so the caching is actually active.

Maybe we need to cache `$HOME/.npm` as well (see [npm cache docs](https://docs.npmjs.com/cli/cache))

**Comment by duk3luk3** [comment]
So `.npm` caching appears to get us nothing - I suppose npm install already doesn't make any web requests at all if it sees packages in `node_modules` that match the version requirements.

I also tried caching brunch, which is taking up a large amount of time too (90s), but that apparently does not support caching at all - https://github.com/brunch/brunch/issues/1379
